lib/models/noSignal.js

```diff
@@ -63,36 +63,63 @@ function alertFunc(nsLineRule, entity) {
         d.exit();
     });
     d.run(function() {
+        logger.debug(context, 'alertfunc nsLineRule %j  entity %j ', nsLineRule, entity);
         // We duplicate info in event and event.ev for VR and non-VR action parameters
         var event = {
             service: nsLineRule[SERVICE],
             subservice: nsLineRule[SUBSERVICE],
             ruleName: nsLineRule[NAME],
             reportInterval: nsLineRule[REPORT_INTERVAL],
-            id: entity._id.id,
-            type: entity._id.type,
             internalCurrentTime: new Date().toISOString()
         };
-
-        // Search for modDate of the entity's attribute
-        // and copy every attribute (if not in event yet)
-        // for use in action template
-        Object.keys(entity.attrs).forEach(function(attrName) {
-            if (attrName === nsLineRule[ATTRIBUTE]) {
+        if (!config.nonSignalByAPI) {
+            // entity is really a entity doc obtained from mongo
+            event.id = entity._id.id;
+            event.type = entity._id.type;
+            logger.debug(context, 'alertfunc event %j ', event);
+            // Search for modDate of the entity's attribute
+            // and copy every attribute (if not in event yet)
+            // for use in action template
+            Object.keys(entity.attrs).forEach(function(attrName) {
+                if (attrName === nsLineRule[ATTRIBUTE]) {
+                    try {
+                        lastTime = new Date(entity.attrs[attrName].modDate * 1000).toISOString();
+                    } catch (ex) {
+                        myutils.logErrorIf(ex, 'run ', d.context);
+                    }
+                }
+                if (event[attrName] === undefined) {
+                    if (entity.attrs[attrName].type === 'DateTime') {
+                        event[attrName] = new Date(entity.attrs[attrName].value * 1000).toISOString();
+                    } else {
+                        event[attrName] = entity.attrs[attrName].value;
+                    }
+                }
+            });
+        } else {
+            // entity is and NGSI object
+            event.id = entity.id;
+            event.type = entity.type;
+            logger.debug(context, 'alertfunc event %j ', event);
+            // Search for modDate of the entity's attribute
+            // and copy every attribute (if not in event yet)
+            // for use in action template
+            const attrName = nsLineRule[ATTRIBUTE];
+            if (entity[attrName]) {
                 try {
-                    lastTime = new Date(entity.attrs[attrName].modDate * 1000).toISOString();
+                    lastTime = entity[attrName].metadata.TimeInstant.value;
```

Comment:
Comment for all: we are checking the attr metadata, but not the modDate... I'm not sure it is correct. Probably is the best we have, but not every attribute will be suitable for this mechanism. We've got an internal _modDate for the entity but no for the attr, right?

Reply:
Fermin clarifies that "modDate" is the built in metadata from MONGO/NGSI and not an arbitrary metadata. OK.

Reply:
As extra info: the name of the metadata is `dateModified` (see https://github.com/telefonicaid/fiware-orion/blob/master/doc/manuals/orion-api.md#builtin-metadata), although in DB the field is named `modDate` (https://github.com/telefonicaid/fiware-orion/blob/master/doc/manuals/admin/database_model.md#entities-collection).